core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -170,7 +170,9 @@ private ManifestFile copyManifest(ManifestFile manifest) {
         specsById,
         newFile,
         snapshotId(),
-        summaryBuilder);
+        summaryBuilder,
+        current.properties().get(TableProperties.AVRO_COMPRESSION),
```

Comment:
what I like about https://github.com/apache/iceberg/pull/8284 is that it just passes a Map of properties rather than the codec + compression level. Maybe we should do the same here to be more future-proof in case we'd want to pass anything else later on?

Reply:
Funnily enough, this is what @sumeetgajjar did initially (see https://github.com/apache/iceberg/pull/6799/commits/62cd446ffc99d83062e2082acb5a26cdfa517c00).
However, he changed it after the initial feedback (see https://github.com/apache/iceberg/pull/6799/commits/7ab1d5c26bddca0e3540b12eeae48a5be3c2cd84).
The feedback that led him to do this is from @rdblue: https://github.com/apache/iceberg/pull/5893#issuecomment-1267261389 (see also https://github.com/apache/iceberg/pull/5893#discussion_r987094412).

Reply:
@nastra please let me know what you think.

Reply:
@nastra shall we follow @rdblue's feedback and keep to using compression codec and level?

Reply:
ah sorry I didn't see that this came up earlier. Then let's go with what @rdblue suggested